js/src/logger.ts

```diff
@@ -1628,6 +1723,100 @@ export class Experiment {
   }
 }
 
+// Format and utils for serializing objects to create spans later. Keep this in
+// sync with the span serialization implementation in logger.py.
+type SpanObjectIds =
+  | { object_kind: "e"; project_id: string; experiment_id: string }
+  | { object_kind: "pl"; org_id: string; project_id: string; log_id: "g" };
+
+type SpanParentSpanIds =
+  | { parent_span_kind: "sub_span"; span_id: string; root_span_id: string }
+  | { parent_span_kind: "root_span"; span_id: string; root_span_id?: undefined }
+  | { parent_span_kind: "none"; span_id?: undefined; root_span_id?: undefined };
+
+type SerializedSpanInfo = SpanObjectIds & SpanParentSpanIds;
+
+function SerializedSpanInfoToString(info: SerializedSpanInfo): string {
+  const objectKindIds = (() => {
+    if (info.object_kind === "e") {
+      return [info.project_id, info.experiment_id];
+    } else if (info.object_kind === "pl") {
```

Comment:
this seems like good logic to write some simple unit tests for

Reply:
done!